core/src/main/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilder.java

```diff
@@ -228,15 +225,13 @@ private Event processBlockingSynchronous(Event event) throws MessagingException,
             } else {
               Event request = createMuleEventForCurrentFlow(event, event.getReplyToDestination(), event.getReplyToHandler());
               sink.accept(request);
-              return Mono.from(request.getContext())
+              return Mono.from(request.getContext().getResponsePublisher())
```

Comment:
take the chance to properly comment this class, specially the bridge that happens on `org.mule.runtime.core.internal.construct.DefaultFlowBuilder.DefaultFlow#apply`